### DISCUSSION TITLE
LibreChat chat history and state get deleted

### DISCUSSION THREAD

**Discussion by NikolaZubic** [danny-avila/LibreChat#4415] [General]
Does anyone know how to save the state when using LibreChat so that when I start localhost again, I do not lose memory (my previous chats and keys set), and even the registered user credentials are not stored? I'm using it with Docker.

**Comment by PrashamTrivedi** [comment]
Stumbled here because I faced similar issue.

If you have any existing data, try running 

```sh
 docker compose down && docker rmi $(docker images | grep librechat | awk '{print $3}') && docker compose
up -d
```

And observe if you got your data back...


Not 100% sure but if you get your data back, there is some sort of lock happening that made the file system inaccessible for the said session. This happened to me and I only lost couple of chats. Got everything back post that
